### ISSUE TITLE
mqtt5 client doesnt receive properties for messages when clean_start = false

### ISSUE THREAD

**Issue by steve999lee** [vernemq/vernemq#1649]
### Environment

- VerneMQ Version: later than1.10.5 (built today)
- OS: Ubuntu
- Erlang/OTP version OTP 22
- VerneMQ configuration (vernemq.conf) or the changes from the default

allow_anonymous = on
listener.tcp.default = 127.0.0.1:10053
listener.tcp.allowed_protocol_versions = 3,4,5
plugins.vmq_passwd = off
plugins.vmq_acl =  on

- Cluster size/standalone: Standalone

### Expected behavior

- When reconnecting a mqtt5 client with clean_start=False, messages queued prior to the client starting should be delivered including their properties.

### Actual behaviour
- Messages are delivered by properties are empty

### Please include minimal steps to reproduce the issue, if possible.
- Build and Deploy broker (default) settings + above config
- Start publisher - publishes message every 2 secs
- Start listener (clean_start = False)
- All good
- Stop listener for a few seconds and then restart
- Listener receives published messages but properties are missing for those sent whilst it was stopped

E.g. 
DEBUG:actions:Received PUBLISH (d0, q1, r0, m35), 'Hello/1295', properties=[], ...  (7 bytes)
DEBUG:actions:Sending PUBACK (Mid: 35)
Hello/1295:0:[]

import paho.mqtt.client as mqtt
import paho.mqtt.properties as properties
import paho.mqtt.packettypes as packettypes
import paho.mqtt.subscribeoptions as subscribeoptions
import time
import logging
logger = logging.getLogger('actions')
logging.warn('Initialising logging')
handler = logging.FileHandler('/var/log/paho.log')
formatter = logging.Formatter('%(asctime)s %(levelname)s %(message)s')
handler.setFormatter(formatter)
logger.addHandler(handler)
logger.setLevel(logging.DEBUG)

```
def on_connect_callback(client, userdata, flags, rc, properties):
    print("Connection attempt returned: " + mqtt.connack_string(rc)) # mqtt !!
    if flags['session present'] == 0:
        print("Session not present")
        client.subscribe("Hello/+", qos=1)
    else:
        print("Session present")

def on_message(client, userdata, msg):
    print("%s:%s:%s" % (msg.topic, str(msg.retain), msg.properties))

clientid="ABCDE"
client = mqtt.Client(clientid, protocol=mqtt.MQTTv5)
client.enable_logger(logger)
client.on_connect = on_connect_callback
client.on_message = on_message
client.enable_logger(logger)
p = properties.Properties(packettypes.PacketTypes.CONNECT)
p.SessionExpiryInterval = 0xFFFFFFFF
client.connect("127.0.0.1", 10053, keepalive=60, clean_start=False, properties=p)
client.loop_forever()
```

**Comment by ioolkos** [comment]
@steve999lee thanks, that's correct, we need to add properties when storing to the Message store: https://github.com/vernemq/vernemq/issues/1560

**Comment by steve999lee** [comment]
@ioolkos - Ah I saw #1560 but it only mentions `user` properties. Looks like it's all properties.

Is anyone working on this or do I need to learn erlang?

**Comment by ioolkos** [comment]
Uptake on v5 is somewhat slow, so probably no hurry. But yeah, we totally need this in at some point, and any help is welcome! 
(learning Erlang is fun :)

**Comment by steve999lee** [comment]
Thanks I'll take a look at the code but I'm worried that you are saying that mqtt5 support is low priority when vernemq claims to support the full mqtt specification.
I may have to revert to mqtt311.

**Comment by ioolkos** [comment]
I understand. What the project helps is probably less worry but more enthousiasm & support ;)

**Comment by steve999lee** [comment]
@ioolkos - could you let me know the procedure for contributing a fix?

**Comment by steve999lee** [comment]
I have a possible fix...I guess I need to push my branch and create a pull request?

**Comment by ioolkos** [comment]
@steve999lee :) 
You need to fork VerneMQ into your Github account, create a branch there, and PR from there to VerneMQ master.

**Comment by briggySmalls** [comment]
Hello! And thanks for this great broker.

Can I please suggest that this issue is reopened. I have encountered this issue with the `erlio/docker-vernemq:1.10.0` docker image. It was very surprising to me to find that the MQTT v5 `CorrelationData` property I was sending is missing if the message had been buffered by the broker (due to `clean_start=true`).

**Comment by ioolkos** [comment]
@briggySmalls persisting v5 properties is only fixed in master, or release 1.12.0

I seem to have forgotten to add this to the changelog though, so thanks

**Comment by briggySmalls** [comment]
I apologise. Yes, as you say, I just tested the latest docker container `vernemq/vernemq:latest` and it indeed works! Thank you for you speedy response and sorry to not do my due dilligence
